src/session_manager.rs

```diff
@@ -729,6 +729,37 @@ impl SessionManager {
         }
     }
 
+    pub(crate) fn handle_telegram_link(&self, link: String) {
+        let client_id = self.active_logged_in_client_id();
+        if let Some(client_id) = client_id {
+            spawn(clone!(@weak self as obj => async move {
```

Comment:
Instead of having all these spawn calls, I think it's better to have this whole function (and `open_chat_by_username`) async and then having the callers of this function to spawn a future to execute it.